src/main/java/com/arpnetworking/metrics/mad/parsers/CollectdJsonToRecordParser.java

```diff
@@ -0,0 +1,389 @@
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Parses Collectd JSON data as a {@link Record}.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public final class CollectdJsonToRecordParser implements Parser<List<DefaultRecord.Builder>> {
+    /**
+     * Parses a collectd POST body.
+     *
+     * @param data byte array of json encoded data
+     * @return A list of {@link DefaultRecord.Builder}
+     * @throws ParsingException if the body is not parsable as collectd formatted json data
+     */
+    @SuppressFBWarnings("SIC_INNER_SHOULD_BE_STATIC_ANON")
+    public List<DefaultRecord.Builder> parse(final byte[] data) throws ParsingException {
+        // Attempt to parse the data as JSON to distinguish between the legacy
+        // format and the current JSON format
+        final JsonNode jsonNode;
+        try {
+            final List<CollectdRecord> records = OBJECT_MAPPER.readValue(data, new TypeReference<List<CollectdRecord>>() { });
+            final List<DefaultRecord.Builder> builders = Lists.newArrayList();
+            for (final CollectdRecord record : records) {
+                final DefaultRecord.Builder builder = new DefaultRecord.Builder();
+                final Map<String, Metric> metrics = Maps.newHashMap();
+                builder.setHost(record.getHost())
+                        .setId(UUID.randomUUID().toString())
+                        .setAnnotations(Collections.emptyMap())
+                        .setTime(record.getTime());
+
+                final String plugin = record.getPlugin();
+                final String pluginInstance = record.getPluginInstance();
+                final String type = record.getType();
+                final String typeInstance = record.getTypeInstance();
+
+                for (final CollectdRecord.Sample sample : record.getSamples()) {
+                    final String metricName = computeMetricName(plugin, pluginInstance, type, typeInstance, sample.getDsName());
+                    final MetricType metricType = mapDsType(sample.getDsType());;
+                    final Metric metric = new DefaultMetric.Builder()
+                            .setType(metricType)
+                            .setValues(Collections.singletonList(
+                                    new Quantity.Builder().setValue(sample.getValue()).build()))
+                            .build();
+                    metrics.merge(metricName, metric, CollectdJsonToRecordParser::mergeMetrics);
+                }
+                builder.setMetrics(metrics);
+                builders.add(builder);
+            }
+            return builders;
+        } catch (final IOException ex) {
+            // CHECKSTYLE.OFF: IllegalInstantiation - Approved for byte[] to String
+            throw new ParsingException(
```

Comment:
This is far from ideal. Can you file an issue to extend ParsingException either its functionality or a new specialization that will accept the buffer (e.g. byte[]).